File: models/Book.java
public class Book {
    private String ID_Book;
    private String Name_Book;
    private String _Author;
    private String _Major;

    public Book(String IDBook, String nameBook, String author, String major) {
        this.ID_Book = IDBook;
        this.Name_Book = nameBook;
        this._Author = author;
        this._Major = major;
    }

    public String getID_Book() {
        return ID_Book;
    }

    public String getName_Book() {
        return Name_Book;
    }

    public String get_Author() {
        return _Author;
    }

    public String get_Major() {
        return _Major;
    }
}
    

File: CSVHandler/book/Book.cpp
// Book.cpp
#include "Book.h"

Book::Book() : bookID(0), name(""), author(""), major("") {
}

Book::Book(const std::string& bookID, const std::string& name, const std::string& author, const std::string& major)
    : bookID(bookID), name(name), author(author), major(major) {
}

std::string Book::getBookID() const {
    return bookID;
}

std::string Book::getName() const {
    return name;
}

std::string Book::getAuthor() const {
    return author;
}

std::string Book::getMajor() const {
    return major;
}

void Book::setBookID(const std::string& bookID) {
    this->bookID = bookID;
}

void Book::setName(const std::string& name) {
    this->name = name;
}

void Book::setAuthor(const std::string& author) {
    this->author = author;
}

void Book::setMajor(const std::string& major) {
    this->major = major;
}
